spectrochempy/analysis/kinetic_utilities.py

```diff
@@ -63,18 +81,22 @@ def _interpret_equation(eq, species):
                 f"Available species : {species}"
             )
         coef = match.group(4) if match.group(4) else 1
-        equation[s] = float(coef) * mult
+        try:
+            coef = int(coef)
+        except ValueError:
+            raise ValueError(
+                f"Stoichiometric coeffcients must be integers. Could not "
+                f"convert {coef} in int"
+            )
+        if is_reactant:
+            left[s] = coef
+        else:
+            right[s] = coef
+
         if match.group(6) in ["->", "→"]:
             # shift to products
-            mult = 1
-
-    # normalize the coefficients (for float we assume that they have at the maximum
-    # 2 decimals
-    y = (np.array(list(equation.values())) * 100).astype(int)
-    div = np.gcd.reduce(np.abs(y))
-    equation = {k: int(v * 100 / div) for k, v in equation.items()}
```

Comment:
normalization is not expected as we are dealing with elementary steps

Reply:
Ok I understand, but what about using a scaling factor to transform into int. so the eq. could be write either way (int or not)?  but ok no problem with this.

Reply:
Actually,  the rate expression for 2 A -> 2B  (r = k C_A**2) differs from that of A -> B  (r = k C_A), so it is safer to not normalize)